mesonbuild/compilers/compilers.py

```diff
@@ -1038,12 +1040,17 @@ def get_pic_args(self):
         m = 'Language {} does not support position-independent code'
         raise EnvironmentException(m.format(self.get_display_language()))
 
-    def get_pie_args(self):
-        m = 'Language {} does not support position-independent executable'
-        raise EnvironmentException(m.format(self.get_display_language()))
+    def get_pie_args(self, enabled: T.Optional[bool]) -> T.List[str]:
+        """Get arguments for pie.
+
+        If enabled is true then the arguments will enable pie explicitly, if
+        it's false they'll disable it. IF it is None, then nothing will be
+        done, and you will get the compiler default.
+        """
+        return []
```

Comment:
Shouldn't you raise exception here if enabled is not None? Otherwise compilers that does not override this method will silently, if I understand correctly.

Reply:
That's intentional, otherwise using a language without pie like D will always error if you set pie, which means you can't mix pie and non-pie languages and not use the default value. Which seems bad. I've added a test case in the latest version for that.